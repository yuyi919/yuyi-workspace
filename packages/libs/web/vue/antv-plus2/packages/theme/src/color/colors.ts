import { getThemeColors } from "./themeColors";

export const baseColors = {
  // color palettes
  blue: "#1890ff",
  purple: "#722ed1",
  cyan: "#13c2c2",
  green: "#52c41a",
  magenta: "#eb2f96",
  pink: "#eb2f96",
  red: "#f5222d",
  orange: "#fa8c16",
  yellow: "#fadb14",
  volcano: "#fa541c",
  geekblue: "#2f54eb",
  lime: "#a0d911",
  gold: "#faad14",
} as const;
export type BaseColorsKey = keyof typeof baseColors;
const Colors = {
  ...getThemeColors("blue", baseColors.blue),
  ...getThemeColors("purple", baseColors.purple),
  ...getThemeColors("cyan", baseColors.cyan),
  ...getThemeColors("green", baseColors.green),
  ...getThemeColors("magenta", baseColors.magenta),
  ...getThemeColors("pink", baseColors.pink),
  ...getThemeColors("red", baseColors.red),
  ...getThemeColors("orange", baseColors.orange),
  ...getThemeColors("yellow", baseColors.yellow),
  ...getThemeColors("volcano", baseColors.volcano),
  ...getThemeColors("geekblue", baseColors.geekblue),
  ...getThemeColors("lime", baseColors.lime),
  ...getThemeColors("gold", baseColors.gold),
} as const;
// @preset-colors: pink, magenta, red, volcano, orange, yellow, gold, cyan, lime, green, blue, geekblue,
//   purple;
export { Colors };
