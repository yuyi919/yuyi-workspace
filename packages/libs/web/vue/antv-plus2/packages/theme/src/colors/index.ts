export { default as common } from './common';
export { default as red } from './red';
export { default as pink } from './pink';
export { default as purple } from './purple';
export { default as deepPurple } from './deepPurple';
export { default as indigo } from './indigo';
export { default as blue } from './blue';
export { default as lightBlue } from './lightBlue';
export { default as cyan } from './cyan';
export { default as teal } from './teal';
export { default as green } from './green';
export { default as lightGreen } from './lightGreen';
export { default as lime } from './lime';
export { default as yellow } from './yellow';
export { default as amber } from './amber';
export { default as orange } from './orange';
export { default as deepOrange } from './deepOrange';
export { default as brown } from './brown';
export { default as grey } from './grey';
export { default as blueGrey } from './blueGrey';

export interface Color {
  50: string;
  100: string;
  200: string;
  300: string;
  400: string;
  500: string;
  600: string;
  700: string;
  800: string;
  900: string;
  A100: string;
  A200: string;
  A400: string;
  A700: string;
}
