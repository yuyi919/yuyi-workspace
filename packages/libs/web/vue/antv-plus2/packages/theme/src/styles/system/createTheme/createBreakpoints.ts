import { OverridableStringUnion } from "../types";

export interface BreakpointOverrides {
  mobile: true;
  tablet: true;
  laptop: true;
  desktop: true;
}

export type BreakpointCore = "xs" | "sm" | "md" | "lg" | "xl";
export type Breakpoint = OverridableStringUnion<BreakpointCore, BreakpointOverrides>;
export const keys: Breakpoint[] = ["xs", "sm", "md", "lg", "xl"];

export type BreakPointRecord = {
  [key in BreakpointCore]: number;
} &
  { [key in Exclude<Breakpoint, BreakpointCore>]?: number };

export interface Breakpoints {
  /**
   * @internal
   */
  unit: string;
  keys: Breakpoint[];
  values: BreakPointRecord;
  up: (key: Breakpoint | number) => string;
  down: (key: Breakpoint | number) => string;
  between: (start: Breakpoint | number, end: Breakpoint | number) => string;
  only: (key: Breakpoint) => string;
}
const defaultBreakPoint = {
  xs: 0, // phone
  sm: 600, // tablets
  md: 900, // small laptop
  lg: 1200, // desktop
  xl: 1536, // large screens
} as BreakPointRecord;
export type BreakpointsOptions = Partial<
  {
    unit: string;
    step: number;
  } & Omit<Breakpoints, "values">
> & {
  values?: Partial<BreakPointRecord>;
};

export function setupBreakpoints(values?: Partial<BreakPointRecord>): BreakPointRecord {
  if (!values) return defaultBreakPoint;
  const result = { ...values } as BreakPointRecord;
  for (const key in values) {
    result[key as Breakpoint] ??= defaultBreakPoint[key as Breakpoint]!;
  }
  return result;
}

export function createBreakpoints(options: BreakpointsOptions): Breakpoints {
  const {
    // The breakpoint **start** at this value.
    // For instance with the first breakpoint xs: [xs, sm).
    values: source,
    unit = "px",
    step = 5,
    ...other
  } = options as BreakpointsOptions;
  const values = setupBreakpoints(source);
  const keys = Object.keys(values) as Breakpoint[];

  function up(key: Breakpoint | number) {
    const value = (
      typeof values[key as Breakpoint] === "number" ? values[key as Breakpoint] : key
    ) as number;
    return `@media (min-width:${value}${unit})`;
  }

  function down(key: Breakpoint | number) {
    const value = (
      typeof values[key as Breakpoint] === "number" ? values[key as Breakpoint] : key
    ) as number;
    return `@media (max-width:${value - step / 100}${unit})`;
  }

  function between(start: Breakpoint | number, end: Breakpoint | number) {
    const endIndex = keys.indexOf(end as Breakpoint);

    return (
      `@media (min-width:${
        typeof values[start as Breakpoint] === "number" ? values[start as Breakpoint] : start
      }${unit}) and ` +
      `(max-width:${
        ((endIndex !== -1 && typeof values[keys[endIndex] as Breakpoint] === "number"
          ? values[keys[endIndex] as Breakpoint]
          : end) as number) -
        step / 100
      }${unit})`
    );
  }

  function only(key: Breakpoint) {
    if (keys.indexOf(key) + 1 < keys.length) {
      return between(key, keys[keys.indexOf(key) + 1]);
    }

    return up(key);
  }

  return {
    keys,
    values: values as BreakPointRecord,
    up,
    down,
    between,
    only,
    unit,
    ...other,
  };
}
