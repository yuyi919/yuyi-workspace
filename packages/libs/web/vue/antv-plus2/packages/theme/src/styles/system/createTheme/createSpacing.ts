import { getPath } from "../utils";

export type SpacingOptions =
  | number
  | Spacing
  | ((abs: number) => number | string)
  | ((abs: number | string) => number | string)
  | ReadonlyArray<string | number>;

export type SpacingArgument = number | string;

// The different signatures imply different meaning for their arguments that can't be expressed structurally.
// We express the difference with variable names.
/* tslint:disable:unified-signatures */
export interface Spacing {
  (): string;
  (value: number): string;
  (topBottom: SpacingArgument, rightLeft: SpacingArgument): string;
  (top: SpacingArgument, rightLeft: SpacingArgument, bottom: SpacingArgument): string;
  (
    top: SpacingArgument,
    right: SpacingArgument,
    bottom: SpacingArgument,
    left: SpacingArgument
  ): string;
}
/* tslint:enable:unified-signatures */

export function createUnaryUnit<Spacing extends SpacingOptions>(
  theme: { spacing: Spacing },
  themeKey?: string,
  defaultValue?: any,
  propName?: string
) {
  const themeSpacing = getPath(theme, themeKey) || defaultValue;

  if (typeof themeSpacing === "number") {
    return (abs: string | number) => {
      if (typeof abs === "string") {
        return abs;
      }

      if (process.env.NODE_ENV !== "production") {
        if (typeof abs !== "number") {
          console.error(
            `Material-UI: Expected ${propName} argument to be a number or a string, got ${abs}.`
          );
        }
      }
      return themeSpacing * abs;
    };
  }

  if (Array.isArray(themeSpacing)) {
    return (abs: string | number) => {
      if (typeof abs === "string") {
        return abs;
      }

      if (process.env.NODE_ENV !== "production") {
        if (!Number.isInteger(abs)) {
          console.error(
            [
              `Material-UI: The \`theme.${themeKey}\` array type cannot be combined with non integer values.` +
                `You should either use an integer value that can be used as index, or define the \`theme.${themeKey}\` as a number.`,
            ].join("\n")
          );
        } else if (abs > themeSpacing.length - 1) {
          console.error(
            [
              `Material-UI: The value provided (${abs}) overflows.`,
              `The supported values are: ${JSON.stringify(themeSpacing)}.`,
              `${abs} > ${themeSpacing.length - 1}, you need to add the missing values.`,
            ].join("\n")
          );
        }
      }

      return themeSpacing[abs];
    };
  }

  if (typeof themeSpacing === "function") {
    return themeSpacing;
  }

  if (process.env.NODE_ENV !== "production") {
    console.error(
      [
        `Material-UI: The \`theme.${themeKey}\` value (${themeSpacing}) is invalid.`,
        "It should be a number, an array or a function.",
      ].join("\n")
    );
  }

  return () => undefined;
}

export function createUnarySpacing<Spacing extends SpacingOptions>(theme: {
  spacing: Spacing;
}): Spacing extends number
  ? (abs: number | string) => number | number
  : Spacing extends any[]
  ? <Index extends number>(abs: Index | string) => Spacing[Index] | string
  : Spacing extends (...args: unknown[]) => unknown
  ? Spacing
  : // warns in Dev
    () => undefined {
  return createUnaryUnit(theme, "spacing", 8, "spacing");
}
export function createSpacing(spacingInput: SpacingOptions = 8): Spacing {
  // Already transformed.
  if ((spacingInput as any).mui) {
    return spacingInput as Spacing;
  }

  // Material Design layouts are visually balanced. Most measurements align to an 8dp grid, which aligns both spacing and the overall layout.
  // Smaller components, such as icons, can align to a 4dp grid.
  // https://material.io/design/layout/understanding-layout.html#usage
  const transform = createUnarySpacing({
    spacing: spacingInput,
  });

  const spacing = (...argsInput: ReadonlyArray<number | string>): string => {
    if (process.env.NODE_ENV !== "production") {
      if (!(argsInput.length <= 4)) {
        console.error(
          `Material-UI: Too many arguments provided, expected between 0 and 4, got ${argsInput.length}`
        );
      }
    }

    const args = argsInput.length === 0 ? [1] : argsInput;

    return args
      .map((argument) => {
        const output = transform(argument);
        return typeof output === "number" ? `${output}px` : output;
      })
      .join(" ");
  };

  spacing.mui = true;

  return spacing;
}
