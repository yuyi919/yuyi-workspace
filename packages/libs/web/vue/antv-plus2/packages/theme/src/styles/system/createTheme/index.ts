export * from "./createTheme";
export * from "./createBreakpoints";
export * from "./createSpacing";
export * from "./shape";
