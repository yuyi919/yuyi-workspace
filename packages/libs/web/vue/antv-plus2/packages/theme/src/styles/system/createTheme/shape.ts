export interface Shape {
  borderRadius: number | string;
}

export type ShapeOptions = Partial<Shape>;

export const shape: Shape = {
  borderRadius: 4,
};
