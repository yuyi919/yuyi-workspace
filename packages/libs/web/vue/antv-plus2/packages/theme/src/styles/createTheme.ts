import { deepmerge } from "./system/utils";
import { createBaseTheme, generateUtilityClass } from "./system";
import { createMixins } from "./createMixins";
import { createPalette } from "./createPalette";
import { createTypography } from "./createTypography";
import { shadows } from "./shadows";
import { createTransitions } from "./createTransitions";
import { zIndex } from "./zIndex";
import type { BaseThemeOptions, BaseTheme } from "./system";
import type { Mixins, MixinsOptions } from "./createMixins";
import type { Palette, PaletteOptions } from "./createPalette";
import type { Typography, TypographyOptions } from "./createTypography";
import type { ShadowsOption } from "./shadows";
import type { Transitions, TransitionsOptions } from "./createTransitions";
import type { ZIndex, ZIndexOptions } from "./zIndex";
import type { Components } from "./components";
import type { Types } from "@yuyi919/shared-types";

export interface ThemeOptions extends BaseThemeOptions {
  mixins?: MixinsOptions;
  components?: Components;
  palette?: PaletteOptions;
  shadows?: ShadowsOption;
  transitions?: TransitionsOptions;
  typography?: TypographyOptions | ((palette: Palette) => TypographyOptions);
  zIndex?: ZIndexOptions;
  unstable_strictMode?: boolean;
}

/**
 * Our [TypeScript guide on theme customization](https://material-ui.com/guides/typescript/#customization-of-theme) explains in detail how you would add custom properties.
 */
export interface Theme extends BaseTheme {
  mixins: Mixins;
  components?: Components;
  palette: Palette;
  shadows: ShadowsOption;
  transitions: Transitions;
  typography: Typography;
  zIndex: ZIndex;
  unstable_strictMode?: boolean;
}
/**
 * Generate a theme base on the options received.
 * @param options Takes an incomplete theme object and adds the missing parts.
 * @param args Deep merge the arguments with the about to be returned theme.
 * @returns A complete, ready to use theme object.
 */
export function createTheme<
  MergeOptions extends Types.Recordable,
  AppendOptions extends Types.Recordable
>(
  options?: MergeOptions & ThemeOptions,
  ...args: AppendOptions[]
): Theme & MergeOptions & AppendOptions {
  const {
    breakpoints: breakpointsInput,
    mixins: mixinsInput = {},
    spacing: spacingInput,
    palette: paletteInput = {},
    shadows: shadowsInput,
    transitions: transitionsInput = {},
    typography: typographyInput = {},
    shape: shapeInput,
    components: componentsInput,
    ...other
  } = options || {};

  const palette = createPalette(paletteInput);
  const systemTheme = createBaseTheme(options);

  let muiTheme: BaseTheme = deepmerge(systemTheme, {
    mixins: createMixins(systemTheme.breakpoints, systemTheme.spacing, mixinsInput),
    palette,
    // Don't use [...shadows] until you've verified its transpiled code is not invoking the iterator protocol.
    shadows: shadowsInput! || shadows.slice(),
    typography: createTypography(palette, typographyInput),
    components: componentsInput || {},
    transitions: createTransitions(transitionsInput),
    zIndex: { ...zIndex },
  });

  muiTheme = deepmerge(muiTheme, other);
  const theme = args.reduce((acc, argument) => deepmerge(acc, argument), muiTheme) as Theme &
    MergeOptions &
    AppendOptions;

  process.env.NODE_ENV !== "production" && productCheck<MergeOptions, AppendOptions>(theme);

  return theme;
}

let warnedOnce = false;

function productCheck<
  MergeOptions extends Types.Recordable,
  AppendOptions extends Types.Recordable
>(theme: Theme & MergeOptions & AppendOptions) {
  const stateClasses = [
    "active",
    "checked",
    "completed",
    "disabled",
    "error",
    "expanded",
    "focused",
    "focusVisible",
    "required",
    "selected",
  ];

  const traverse = (styleOverrides: any, component: string) => {
    let key;

    // eslint-disable-next-line guard-for-in, no-restricted-syntax
    for (key in styleOverrides) {
      const child = styleOverrides[key];
      if (stateClasses.indexOf(key) !== -1 && Object.keys(child).length > 0) {
        if (process.env.NODE_ENV !== "production") {
          const stateClass = generateUtilityClass("", key);
          console.error(
            [
              `Material-UI: The \`${component}\` component increases ` +
                `the CSS specificity of the \`${key}\` internal state.`,
              "You can not override it like this: ",
              JSON.stringify(styleOverrides, null, 2),
              "",
              `Instead, you need to use the '&.${stateClass}' syntax:`,
              JSON.stringify(
                {
                  root: {
                    [`&.${stateClass}`]: child,
                  },
                },
                null,
                2
              ),
              "",
              "https://material-ui.com/r/state-classes-guide",
            ].join("\n")
          );
        }
        // Remove the style to prevent global conflicts.
        styleOverrides[key] = {};
      }
    }
  };

  (Object.keys(theme.components!) as (keyof Components)[]).forEach((component) => {
    //@ts-ignore
    const styleOverrides = theme.components![component]!.styleOverrides!;

    if (styleOverrides && component.indexOf("Mui") === 0) {
      traverse(styleOverrides, component);
    }
  });
}

/**
 * @deprecated
 * Use `import { createTheme } from '@material-ui/core/styles'` instead.
 */
export function createMuiTheme(options?: ThemeOptions, ...args: object[]): Theme;
export function createMuiTheme(...args: [object?, ...object[]]): Theme {
  if (process.env.NODE_ENV !== "production") {
    if (!warnedOnce) {
      warnedOnce = true;
      console.error(
        [
          "Material-UI: the createMuiTheme function was renamed to createTheme.",
          "",
          "You should use `import { createTheme } from '@material-ui/core/styles'`",
        ].join("\n")
      );
    }
  }

  return createTheme(...args);
}

// console.log('createTheme', createTheme({
//   palette: {
//     success: {
//       main: "rgba(1,1,1,1)"
//     }
//   }
// }));

export function createMuiStrictModeTheme<
  MergeOptions extends Types.Recordable,
  AppendOptions extends Types.Recordable
>(
  options?: ThemeOptions & MergeOptions,
  ...args: AppendOptions[]
): Theme & MergeOptions & AppendOptions {
  return createTheme<MergeOptions, AppendOptions>(
    (options
      ? {
          ...options,
          unstable_strictMode: true,
        }
      : {
          unstable_strictMode: true,
        }) as MergeOptions & ThemeOptions,
    ...args
  );
}
