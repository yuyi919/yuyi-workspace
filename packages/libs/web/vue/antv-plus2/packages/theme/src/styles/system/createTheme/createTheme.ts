import { Types } from "@yuyi919/shared-types";
import { deepmerge } from "../utils";
import { Breakpoints, BreakpointsOptions, createBreakpoints } from "./createBreakpoints";
import { createSpacing, Spacing, SpacingOptions } from "./createSpacing";
import { Shape, shape, ShapeOptions } from "./shape";

export type { Breakpoint, BreakpointOverrides } from "./createBreakpoints";

export type Direction = "ltr" | "rtl";

export interface IBasePalette {
  mode: "light" | "dark";
}
export interface BaseThemeOptions {
  shape?: ShapeOptions;
  breakpoints?: BreakpointsOptions;
  direction?: Direction;
  palette?: Partial<IBasePalette>;
  spacing?: SpacingOptions;
  zIndex?: Types.Recordable<number | undefined | null>;
}
export interface BaseTheme {
  shape: Shape;
  breakpoints: Breakpoints;
  direction: Direction;
  palette: IBasePalette;
  spacing: Spacing;
}

/**
 * Generate a theme base on the options received.
 * @param options Takes an incomplete theme object and adds the missing parts.
 * @param args Deep merge the arguments with the about to be returned theme.
 * @returns A complete, ready to use theme object.
 */
export function createBaseTheme(options?: BaseThemeOptions, ...args: object[]): BaseTheme {
  const {
    breakpoints: breakpointsInput = {},
    palette: paletteInput = {},
    spacing: spacingInput,
    shape: shapeInput = {},
    ...other
  } = options || {};

  const breakpoints = createBreakpoints(breakpointsInput);
  const spacing = createSpacing(spacingInput);

  let muiTheme = deepmerge(
    {
      breakpoints,
      direction: "ltr",
      palette: { mode: "light", ...paletteInput },
      spacing,
      shape: { ...shape, ...shapeInput },
    },
    other
  ) as BaseTheme;

  muiTheme = args.reduce((acc: BaseTheme, argument) => deepmerge(acc, argument), muiTheme);

  return muiTheme;
}
