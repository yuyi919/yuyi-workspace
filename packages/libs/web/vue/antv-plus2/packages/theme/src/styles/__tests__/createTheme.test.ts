import { deepOrange, green } from "../../colors";
import { createTheme, Theme } from "../createTheme";

describe("createTheme", () => {
  it("should have a palette", () => {
    const theme = createTheme();
    expect(typeof createTheme).toBe("function");
    expect(typeof theme.palette).toBe("object");
  });

  it("should have the custom palette", () => {
    const theme = createTheme({
      palette: { primary: { main: deepOrange[500] }, secondary: { main: green.A400 } },
    });
    expect(theme.palette.primary.main).toBe(deepOrange[500]);
    expect(theme.palette.secondary.main).toBe(green.A400);
  });

  it("should allow providing a partial structure", () => {
    const theme = createTheme({ transitions: { duration: { shortest: 150 } } });
    expect(theme.transitions.duration.shorter).not.toBe(undefined);
  });

  describe("shadows", () => {
    it("should provide the default array", () => {
      const theme = createTheme();
      expect(theme.shadows[2]).toBe(
        "0px 3px 1px -2px rgba(0,0,0,0.2),0px 2px 2px 0px rgba(0,0,0,0.14),0px 1px 5px 0px rgba(0,0,0,0.12)"
      );
    });

    it("should override the array as expected", () => {
      const shadows = [
        "none",
        1,
        1,
        1,
        2,
        3,
        3,
        4,
        5,
        5,
        6,
        6,
        7,
        7,
        7,
        8,
        8,
        8,
        9,
        9,
        10,
        10,
        10,
        11,
        11,
      ] as const;
      const theme = createTheme({ shadows });
      expect(theme.shadows).toBe(shadows);
    });
  });

  describe("components", () => {
    it("should have the components as expected", () => {
      const components = {
        MuiDialog: {
          defaultProps: {
            fullScreen: true,
            fullWidth: false,
          },
        },
        MuiButtonBase: {
          defaultProps: {
            disableRipple: true,
          },
        },
        MuiPopover: {
          defaultProps: {
            container: null, //document.createElement("div"),
          },
        },
      };
      const theme = createTheme({ components });
      expect(theme.components).toEqual(components);
    });
  });

  describe("styleOverrides", () => {
    it("should warn when trying to override an internal state the wrong way", () => {
      let theme: Theme = null as any;

      expect(() => {
        theme = createTheme({
          components: { Button: { styleOverrides: { disabled: { color: "blue" } } } },
        });
      }).not.toErrorDev();
      // @ts-expect-error
      expect(Object.keys(theme.components!.Button!.styleOverrides!.disabled).length).toBe(1);

      expect(() => {
        theme = createTheme({
          components: { MuiButton: { styleOverrides: { root: { color: "blue" } } } },
        });
      }).not.toErrorDev();

      expect(() => {
        theme = createTheme({
          components: { MuiButton: { styleOverrides: { disabled: { color: "blue" } } } },
        });
      }).toErrorDev(
        "Material-UI: The `MuiButton` component increases the CSS specificity of the `disabled` internal state."
      );
      expect(Object.keys(theme.components!.MuiButton!.styleOverrides!.disabled).length).toBe(0);

      expect(theme.components).toMatchSnapshot();
    });
  });

  it("shallow merges multiple arguments", () => {
    const theme = createTheme({ foo: "I am foo" }, { bar: "I am bar" });
    expect(theme.foo).toBe("I am foo");
    expect(theme.bar).toBe("I am bar");
  });

  it("deep merges multiple arguments", () => {
    const theme = createTheme({ custom: { foo: "I am foo" } }, { custom: { bar: "I am bar" } });
    expect(theme.custom.foo).toBe("I am foo");
    expect(theme.custom.bar).toBe("I am bar");
  });
});
