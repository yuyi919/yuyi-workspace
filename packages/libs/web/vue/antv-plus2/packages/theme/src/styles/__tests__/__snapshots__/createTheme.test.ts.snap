// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`createTheme styleOverrides should warn when trying to override an internal state the wrong way 1`] = `
Object {
  "MuiButton": Object {
    "styleOverrides": Object {
      "disabled": Object {},
    },
  },
}
`;
